language/proto/testdata/file_mode/foo.proto

```diff
@@ -0,0 +1,3 @@
+syntax = "proto3";
+
+package file_mode;
```

Comment:
Can you have one of these import from the other to confirm that the dep detection logic works

Reply:
I have added an import statement to this file, which could test that I haven't broken the parsing logic, but fwiw these tests are just for the generate behavior, not resolve. I haven't made any changes to the resolve behavior, which is tested separately.

Right now I don't believe that resolve is affected by mode changes at all. I could loop through all the available modes on the config here to test that the existing tests pass with all modes? https://github.com/bazelbuild/bazel-gazelle/blob/b1e37463c2ef0997c5cf0e90cdf03e696caaed27/language/proto/resolve_test.go#L357

What do you think?

Reply:
Re: your comment in Slack
> I'm not concerned about it changing the result mode, I want to make sure that it actually does the resolve correctly when there is single file target generation

I've added a simple test that imports a single proto given an index of three different protos with separate `proto_library`s.

The previous test that I added for the comment above has been removed.

Should be good now? @achew22